### PR TITLE
[validator] Correct APIs

### PR DESCRIPTION
**Pull request by TonyPythoneer**
# Improvement to existing type definition

Refer to [validator.js](https://github.com/chriso/validator.js), I added more complete types for `isAlpha` and `isAlphanumeric`.

### CODE REVIEW

**Comment by TonyPythoneer** [comment]
@IOAyman Would you review this? :smile: 
It's based on your #11372 PR, I write more detail for this definition.

File: validator/validator-tests.ts [hunk 1]
```diff
@@ -19,8 +19,87 @@ let any: any;
   result = validator.isAfter('sample', new Date().toString());
 
   result = validator.isAlpha('sample');
-
-  result = validator.isAlphanumeric('sample');
+  result = validator.isAlphanumeric('sample', 'ar');
```
Comment (IOAyman) [inline validator/validator-tests.ts]: These lines of tests (22-60)  should be `.isAlpha(..)`, not `.isAlphanumeric()`

File: validator/validator.d.ts [hunk 2]
```diff
@@ -4,6 +4,9 @@
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
 declare namespace ValidatorJS {
+  type Alpha = "ar" | "ar-AE" | "ar-BH" | "ar-DZ" | "ar-EG" | "ar-IQ" | "ar-JO" | "ar-KW" | "ar-LB" | "ar-LY" | "ar-MA" | "ar-QA" | "ar-QM" | "ar-SA" | "ar-SD" | "ar-SY" | "ar-TN" | "ar-YE" | "cs-CZ" | "de-DE" | "en-AU" | "en-GB" | "en-HK" | "en-IN" | "en-NZ" | "en-US" | "en-ZA" | "en-ZM" | "es-ES" | "fr-FR" | "hu-HU" | "nl-NL" | "pl-PL" | "pt-BR" | "pt-PT" | "ru-RU" | "sr-RS" | "sr-RS@latin" | "tr-TR";
```
Comment (IOAyman) [inline validator/validator.d.ts]: I think is would be better to name this type something like `LocalAlpha`, because that's what it is :)

**Review by IOAyman — CHANGES_REQUESTED** [review]

**Comment by TonyPythoneer** [comment]
Great suggestion! I will fix it.

**Comment by TonyPythoneer** [comment]
@IOAyman, I had fixed.
The `validator.d.ts` changes much because my sublime has typescript auto format. The main change of this PR is still for `isAlpha` and `isAlphanumeric`

**Comment by TonyPythoneer** [comment]
I modified the wrong branch.  I gave up this branch. Close.

File: node/node-tests.ts [hunk 3]
```diff
@@ -1151,6 +1151,228 @@ namespace net_tests {
         // Make sure .listen() and .close() retuern a Server instance
         net.createServer().listen(0).close().address();
     }
+
```
Comment (IOAyman) [inline node/node-tests.ts]: I cannot approve the changes to this file. Please create another PR for this.

File: node/node.d.ts [hunk 4]
```diff
@@ -1666,6 +1666,78 @@ declare module "net" {
         end(str: string, cb?: Function): void;
         end(str: string, encoding?: string, cb?: Function): void;
         end(data?: any, encoding?: string): void;
+
```
Comment (IOAyman) [inline node/node.d.ts]: I cannot approve the changes to this file. Please create another PR for this.

File: validator/validator.d.ts [hunk 5]
```diff
@@ -4,298 +4,306 @@
...
 declare namespace ValidatorJS {
-  interface ValidatorStatic {
+    type AlphaLocale = "ar" | "ar-AE" | "ar-BH" | "ar-DZ" | "ar-EG" | "ar-IQ" | "ar-JO" | "ar-KW" | "ar-LB" | "ar-LY" | "ar-MA" | "ar-QA" | "ar-QM" | "ar-SA" | "ar-SD" | "ar-SY" | "ar-TN" | "ar-YE" | "cs-CZ" | "de-DE" | "en-AU" | "en-GB" | "en-HK" | "en-IN" | "en-NZ" | "en-US" | "en-ZA" | "en-ZM" | "es-ES" | "fr-FR" | "hu-HU" | "nl-NL" | "pl-PL" | "pt-BR" | "pt-PT" | "ru-RU" | "sr-RS" | "sr-RS@latin" | "tr-TR";
+    type AlphanumericLocale = "ar" | "ar-AE" | "ar-BH" | "ar-DZ" | "ar-EG" | "ar-IQ" | "ar-JO" | "ar-KW" | "ar-LB" | "ar-LY" | "ar-MA" | "ar-QA" | "ar-QM" | "ar-SA" | "ar-SD" | "ar-SY" | "ar-TN" | "ar-YE" | "cs-CZ" | "de-DE" | "en-AU" | "en-GB" | "en-HK" | "en-IN" | "en-NZ" | "en-US" | "en-ZA" | "en-ZM" | "es-ES" | "fr-FR" | "fr-BE" | "hu-HU" | "nl-BE" | "nl-NL" | "pl-PL" | "pt-BR" | "pt-PT" | "ru-RU" | "sr-RS" | "sr-RS@latin" | "tr-TR";
+    interface ValidatorStatic {
 
-    // **************
-    // * Validators *
-    // **************
+        // **************
```
Comment (IOAyman) [inline validator/validator.d.ts]: Please only commit the lines you have changed (remove these spaces)

**Review by IOAyman — COMMENTED** [review]
@TonyPythoneer, Thank you for you contribution :smile: 
Only a couple of things and we're good to go :smile:

**Pull request closed without merging**